src/test_common.rs

```diff
@@ -86,9 +76,64 @@ impl Element for [u8; 16] {
     }
 }
 
-pub fn get_vec_tree_from_slice<U: Unsigned>(
+/// Implementation of SHA-256 Hasher
+///
+/// It is used for testing purposes
+#[derive(Copy, Clone)]
+pub struct Sha256Hasher {
+    engine: Sha256,
+}
+
+impl Sha256Hasher {
+    pub fn new() -> Sha256Hasher {
+        Sha256Hasher {
+            engine: Sha256::new(),
+        }
+    }
+}
+
+impl Debug for Sha256Hasher {
+    fn fmt(&self, _f: &mut Formatter<'_>) -> fmt::Result {
+        todo!()
+    }
+}
+
+impl Default for Sha256Hasher {
+    fn default() -> Self {
+        Sha256Hasher::new()
+    }
+}
+
+impl Hasher for Sha256Hasher {
+    // FIXME: contract is broken by design
+    fn finish(&self) -> u64 {
+        unimplemented!()
```

Comment:
FYI: `unimplemented()` also supports adding a message, so perhaps putting the information from the comment into the `unimplemented!()` could make sense.

Reply:
Ok! Added in https://github.com/filecoin-project/merkletree/pull/97/commits/45019d5d5727e5211e2c8c979958eb1a52f6f61e